### ISSUE TITLE
entropy regularisation in sbcfm was not used in the notebook examples

### ISSUE THREAD

**Issue by J-zin** [atong01/conditional-flow-matching#49]
Hi, thanks for your excellent paper and code. It seems that you didn't use the entropy regularisation for sbcfm in the notebook examples. For example, in [training-8gaussians-to-moons.ipynb](https://github.com/atong01/conditional-flow-matching/blob/main/examples/notebooks/training-8gaussians-to-moons.ipynb):

```python
FM = SchrodingerBridgeConditionalFlowMatcher(sigma=sigma)
```

```python
class SchrodingerBridgeConditionalFlowMatcher(ConditionalFlowMatcher):
    """Child class for Schrödinger bridge conditional flow matching method. This class implements
    the SB-CFM methods from [1] and inherits the ConditionalFlowMatcher parent class.

    It overrides the compute_sigma_t, compute_conditional_flow and
    sample_location_and_conditional_flow functions.
    """

    def __init__(self, sigma: float = 1.0, ot_method="exact"):
        r"""Initialize the SchrodingerBridgeConditionalFlowMatcher class. It requires the hyper-
        parameter $\sigma$ and the entropic OT map.

        Parameters
        ----------
        sigma : float
        ot_sampler: exact OT method to draw couplings (x0, x1) (see Eq.(17) [1]).
        """
        self.sigma = sigma
        self.ot_method = ot_method
        self.ot_sampler = OTPlanSampler(method=ot_method, reg=2 * self.sigma**2)
```

The `SchrodingerBridgeConditionalFlowMatcher` has `ot_method="exact"` by default. That means sbcfm exactly solves OT without entropy regularisation. Should we use `SchrodingerBridgeConditionalFlowMatcher(sigma=sigma, method='sinkhorn')`, rather than `SchrodingerBridgeConditionalFlowMatcher(sigma=sigma)`?

**Comment by J-zin** [comment]
ahh, sorry for the confusion. My question actually is: the  `SchrodingerBridgeConditionalFlowMatcher ` class doesn't use the entropy regularisation by default, unless you manually set `ot_method != ''exact`.

the `ot_sampler` in `SchrodingerBridgeConditionalFlowMatcher ` is defined as
```python
self.ot_sampler = OTPlanSampler(method=ot_method, reg=2 * self.sigma**2)
```
Note that the `ot_method` here is 'exact' by default and the `OTPlanSampler` is 
```python
class OTPlanSampler:
    """OTPlanSampler implements sampling coordinates according to an OT plan (wrt squared Euclidean
    cost) with different implementations of the plan calculation."""

    def __init__(
        self,
        method: str,
        reg: float = 0.05,
        reg_m: float = 1.0,
        normalize_cost=False,
        **kwargs,
    ):
        # ot_fn should take (a, b, M) as arguments where a, b are marginals and
        # M is a cost matrix
        if method == "exact":
            self.ot_fn = pot.emd
        elif method == "sinkhorn":
            self.ot_fn = partial(pot.sinkhorn, reg=reg)
        elif method == "unbalanced":
            self.ot_fn = partial(pot.unbalanced.sinkhorn_knopp_unbalanced, reg=reg, reg_m=reg_m)
        elif method == "partial":
            self.ot_fn = partial(pot.partial.entropic_partial_wasserstein, reg=reg)
        else:
            raise ValueError(f"Unknown method: {method}")
        self.reg = reg
        self.reg_m = reg_m
        self.normalize_cost = normalize_cost
        self.kwargs = kwargs
```
That means the `ot_fn` is not an entropy-regularised OT but an exact OT even though you use `reg=2 * self.sigma**2`.

**Comment by kilianFatras** [comment]
@J-zin,

You are right. Theoretically, we should use entropic OT instead of exact OT in the SBCFM class. 

The reason we do not is that, as Alex mentioned, minibatch OT already acts like an entropic regularization on the transport plan (i.e., it densifies the plan). Therefore, if we use both minibatch OT with entropic OT, the plan becomes too close to uniform and we lose all geometric information. This was studied in [1]. This is also one of the reasons in practice we found that minibatch OT performed better than minibatch entropic OT. Therefore, we have set the default to be minibatch exact OT in order to match the best results we got.

I hope this clarifies the issue. If it does, feel free to close it pls.

[1] Learning with minibatch Wasserstein, AISTATS, Fatras et al.
